src/test/java/com/amazon/opendistroforelasticsearch/security/system_indices/SystemIndicesTests.java

```diff
@@ -0,0 +1,547 @@
+        assertEquals(RestStatus.OK.getStatus(), response.getStatusCode());
+
+        XContentParser xcp = XContentType.JSON.xContent().createParser(NamedXContentRegistry.EMPTY, LoggingDeprecationHandler.INSTANCE, response.getBody());
+        SearchResponse searchResponse = SearchResponse.fromXContent(xcp);
+        assertEquals(RestStatus.OK, searchResponse.status());
+        assertEquals(expectecdHits, searchResponse.getHits().getHits().length);
+        assertEquals(0, searchResponse.getFailedShards());
+        assertEquals(5, searchResponse.getSuccessfulShards());
+    }
+
+    @Test
+    public void testSearchAsSuperAdmin() throws Exception {
+        setupSystemIndicesDisabledWithSsl();
+        createTestIndicesAndDocs();
+        RestHelper restHelper = keyStoreRestHelper();
+
+        //search system indices
+        for (String index : listOfIndexesToTest) {
+            validateSearchResponse(restHelper.executePostRequest(index + "/_search", matchAllQuery), 1);
+        }
+
+        //search all indices
+        RestHelper.HttpResponse response = restHelper.executePostRequest("/_search", matchAllQuery);
+        assertEquals(RestStatus.OK.getStatus(), response.getStatusCode());
+    }
+
+    @Test
+    public void testSearchAsAdmin() throws Exception {
+        setupSystemIndicesDisabledWithSsl();
+        createTestIndicesAndDocs();
+        RestHelper restHelper = sslRestHelper();
+
+        //search system indices
+        for (String index : listOfIndexesToTest) {
+            validateSearchResponse(restHelper.executePostRequest(index + "/_search", matchAllQuery, allAccessUserHeader), 1);
+        }
+
+        //search all indices
+        RestHelper.HttpResponse response = restHelper.executePostRequest("/_search", matchAllQuery, allAccessUserHeader);
+        assertEquals(RestStatus.OK.getStatus(), response.getStatusCode());
+    }
+
+    @Test
+    public void testSearchWithSystemIndicesAsSuperAdmin() throws Exception {
+        setupSystemIndicesEnabledWithSsl();
+        createTestIndicesAndDocs();
+        RestHelper restHelper = keyStoreRestHelper();
+
+        //search system indices
+        for (String index : listOfIndexesToTest) {
+            validateSearchResponse(restHelper.executePostRequest(index + "/_search", matchAllQuery), 1);
+        }
+
+        //search all indices
+        RestHelper.HttpResponse response = restHelper.executePostRequest("/_search", matchAllQuery);
+        assertEquals(RestStatus.OK.getStatus(), response.getStatusCode());
+    }
+
+
+    @Test
+    public void testSearchWithSystemIndicesAsAdmin() throws Exception {
+        setupSystemIndicesEnabledWithSsl();
+        createTestIndicesAndDocs();
+        RestHelper restHelper = sslRestHelper();
+
+        for (String index : listOfIndexesToTest) {
+            validateSearchResponse(restHelper.executePostRequest(index + "/_search", matchAllQuery, allAccessUserHeader), 0);
```

Comment:
why are we allowing admin to search system indices? 

System indices will be accessible through es APIs only for AdminDN.  All the other user including admin will be able to access index only through APIs defined by respective plugin.

Reply:
No, we are not allowing `admin` to search system indices. `admin` requests return 200 but with empty results. This is same behavior as security index. This allows `_all\_search` request work, but filters results from system indices.

Reply:
Thanks for clarifying!